### DISCUSSION TITLE
Atributo target

### DISCUSSION THREAD

**Discussion by vitor-mendrone-deel** [frontendbr/forum#81] [Discussão]
Pessoal qual a opinião de você sobre o uso do atributo target="_blank"?

Já li alguns artigos sobre e a maioria dizia que era bom evitar o uso inclusive por questões de acessibilidade. 

Como trabalho com E-commerce sou muito cobrado para inseri-lo em todos os links externos pois os lojistas não querem que a pessoa saia da loja para abrir um link.

**Comment by bernardodiasc** [comment]
Opinião pessoal: se vai quebrar o fluxo da aplicação, abre outra pagina, por exemplo links externos. Se faz parte do fluxo, por exemplo um link interno que voce quer que o usuario siga, abre na mesma pagina.

**Comment by dionarodrigues** [comment]
É mamílico esse tema. Para o negócio, talvez pode ser melhor abrir em outra aba. No entanto, penso que é melhor deixar o usuário decidir isso.

**Comment by bernardodiasc** [comment]
Há controversias mesmo, se pensar demais sobre o que o usuario quer, talvez o usuario nao goste das cores e estilos e funcionalidades que voce escolheu pro site, entao é melhor nem usar css e js... hehehehe brincadeira. Só colocando lenha na fogueira mesmo. Acho que muitas decisoes nós desenvolvedores temos que tomar, é nosso trabalho. O caso de remover a habilidade de zoom por exemplo é ruim, mas abrir links externos em nova aba acho bom. Falando isso pois tem que parar e pensar mesmo sobre o projeto e como ele funciona melhor, nao acredito que 1 resposta valha pra todos os casos, cada caso é um caso... :)

**Comment by felipefialho** [comment]
Puts, polemico mesmo né?

As boas praticas sempre disseram para deixar o usuário decidir o que fazer (ou no caso, abrir o link em uma nova aba, caso queira), porém um dia desses escutei algo que me fez pensar um pouco diferente:

> Nem todos os usuários do seu produto são heavy users de tecnologia e sabem abrir facilmente o link em uma nova aba.

Então, como disse o @bernardodiasc, em tese, abrir links que quebram o fluxo de navegação em uma nova aba faz todo sentido. Colocando-se na pele de um usuário leigo, com certeza é extremamente chato você clicar em um link e ele abrir um site externo na mesma aba.

**Comment by dionarodrigues** [comment]
Entendo vocês. E, em partes, concordo também.

Mas e se o usuário desejar que o link seja aberto na mesma aba? Não há como ele fazer isso. O que pode ser frustrante em algumas situações. Já o contrário, pode ser feito e decidido entre duas opções.  Hehehe

**Comment by bernardodiasc** [comment]
Faz muito sentido isso, realmente. Como disse no meu primeiro comentário, opinião pessoal, alguns casos não me importa o que o usuário quer, contrariando boas práticas, retiro as opções intencionalmente. Acho que fazemos isso o tempo todo em várias outras ocasiões. O caso de links externos é o único que me vem em mente agora para o target blank. :)

**Comment by felipefialho** [comment]
@diogorodrigues Faz todo sentido do mundo, mas acho extremamente improvável que o usuário queira abrir um site externo na mesma aba =p

**Comment by vitor-mendrone-deel** [comment]
Obrigado pessoal!

Penso como vocês, se o link quebra o fluxo da aplicação é melhor que seja aberto externamente.

A verdade é que a maior parte dos usuários não sabe bem como abrir um link em uma nova aba. É aquele negócio, vivemos em um país onde mais de 80% das pessoas tem dificuldades de comunicação, é complicado esperar que os usuários saibam abrir um link em uma nova aba quando mal sabem interpretar um texto. 

Então mesmo indo contra as boas práticas de forma global, em se tratando do nosso país é sempre uma boa ideia considerar que o usuário tem o mínimo conhecimento possível de comportamentos da máquina.

**Comment by bernardodiasc** [comment]
@Mendrone 80% é um numero inventado certo? ;)

Não é um problema do país, o lance é que não podemos assumir que todo mundo no mundo tem o mesmo nível técnico ou sequer gosta de computadores. Por isso as vezes tomamos as decisões.

**Comment by vitor-mendrone-deel** [comment]
Infelizmente não, @bernardodiasc 

http://oglobo.globo.com/sociedade/educacao/so-8-dos-brasileiros-sabem-se-expressar-plenamente-diz-pesquisa-18773598

Mas sim, não é bem um problema do país mas é um problema maior aqui (e em outros países do mundo com problemas semelhantes na educação)

**Comment by vitor-mendrone-deel** [comment]
Vale a pena dar uma olhada nessa issue do @fdaciuk  #87
